opentelemetry/spring-boot-instrumentation/src/test/java/com/example/demo/DockerComposeTestsIT.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.demo;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.client.util.Preconditions;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.regex.Pattern;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.testcontainers.containers.ComposeContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+public class DockerComposeTestsIT {
+  @ClassRule
+  public static ComposeContainer environment =
+      new ComposeContainer(new File("docker-compose.yaml"), new File("docker-compose.adc.yaml"))
+          .withEnv("USERID", System.getenv("USERID"))
+          .withEnv("GOOGLE_CLOUD_PROJECT", getenvNotNull("GOOGLE_CLOUD_PROJECT"))
+          .withEnv(
+              "GOOGLE_APPLICATION_CREDENTIALS", getenvNotNull("GOOGLE_APPLICATION_CREDENTIALS"))
+          .withExposedService("app", 8080)
+          .withExposedService("otelcol", 8888)
+          .waitingFor("app", Wait.forHttp("/multi"))
+          .withTailChildContainers(true)
+          .withBuild(true);
+
+  @Test
+  public void testApp() throws InterruptedException, IOException, URISyntaxException {
+    // Let the docker compose app run until some spans/logs/metrics are sent to
+    // GCP
+    Thread.sleep(60_000);
```

Comment:
nit: is it possible to define the timeout for the test? can we decrease 1 minute run of the test?